test/components/preprocessors/test_recursive_splitter.py

```diff
@@ -990,3 +990,29 @@ def test_run_complex_text_with_multiple_separators():
     assert len(chunks[3].content) == 152
     assert chunks[3].content.startswith("C")
     assert chunks[3].content.endswith("D" * 50)
+
+
+def test_recursive_splitter_generates_unique_ids_and_correct_meta():
+    """
+    • Every produced chunk must have a unique `id`.
+    • Each chunk must carry `parent_id` = original doc’s ID.
```

Comment:
I think this docstring is not strictly needed (the test name and the comments are already descriptive enough).